### PR TITLE
Improved NamedTuple declaration

### PR DESCRIPTION
**Pull request by amogus07**
## Description

Utilize a [new way of declaring NamedTuples](https://docs.python.org/3/library/typing.html#typing.NamedTuple), which allows for typechecking as well. Maybe the latter is now redundant in other places, but I'm not that familiar with the codebase yet, so I just changed the declarations (and hopefully used the correct types). While I was at it, I also ran `poetry update`, but I'll revert poetry.lock in case I wasn't supposed to do that. This is my first commit here, so I hope I didn't do anything wrong...

## To Do

- [x] ~Documentation.~
- [x] ~Changelog.~
- [ ] Tests.

### CODE REVIEW

File: poetry.lock [hunk 1]
```diff
@@ -74,13 +74,13 @@ test = ["tox"]
 
 [[package]]
 name = "babel"
-version = "2.15.0"
+version = "2.16.0"
```
Comment (snejus) [inline poetry.lock:77]: Presumable this update was not intentional? It would have to be done in a separate PR on its own

**Review by snejus — CHANGES_REQUESTED** [review]
I like this update!

Just checked and it seems like there's one replacements that's been left out?

image

**Comment by amogus07** [comment]
uhhhh… can't reproduce this locally?!

**Comment by amogus07** [comment]
@snejus What should I do now? Can you reproduce it?

**Comment by snejus** [comment]
`dict` is only supported from Python 3.9 onwards, so try using `typing.Dict` instead. Also, I think you may want to use `str` type for the key instead of `Any`, `Dict[str, Any]`.

File: beets/dbcore/db.py [hunk 2]
```diff
@@ -322,7 +322,7 @@ class Model(ABC):
     """
 
     @cached_classproperty
-    def _relation(cls) -> type[Model]:
+    def _relation(cls) -> Model:
```
Comment (snejus) [inline beets/dbcore/db.py:325]: Was this change intentional? This needs to be `type[Model]`

File: beets/plugins.py [hunk 3]
```diff
@@ -291,7 +292,7 @@ def load_plugins(names=()):
             )
 
 
-_instances = {}
+_instances: Dict[Any, Any] = {}
```
Comment (snejus) [inline beets/plugins.py:295]: I think it should be possible to specify accurate types for both the key and value here

File: beets/ui/commands.py [hunk 4]
```diff
@@ -665,7 +664,7 @@ def show_match_tracks(self):
         """
         # Tracks.
         # match is an AlbumMatch named tuple, mapping is a dict
-        # Sort the pairs by the track_info index (at index 1 of the namedtuple)
+        # Sort the pairs by the track_info index (at index 1 of the NamedTuple)
```
Comment (snejus) [inline beets/ui/commands.py:667]: If you're adjusting this, could you clarify what **at index 1 of the NamedTuple** means? Can this be replaced by the name of the namedtuple attribute?

File: beets/util/__init__.py [hunk 5]
```diff
@@ -833,7 +838,7 @@ def plurality(objs: Sequence[T]) -> T:
     return c.most_common(1)[0]
 
 
-def convert_command_args(args: List[bytes]) -> List[str]:
+def convert_command_args(args: Sequence[Bytes_or_String]) -> List[str]:
```
Comment (snejus) [inline beets/util/__init__.py:841]: The type fixes in this module are unfortunately out of scope of the `namedtuple` -> `NamedTuple` replacement. Please undo them and submit a separate PR as they need to be reviewed carefully.

**Review by snejus — CHANGES_REQUESTED** [review]
Ideally we want to keep this PR focused on the namedtuple replacement, so I asked to submit the rest of the type changes as a separate PR.

File: beets/autotag/match.py [hunk 6]
```diff
@@ -78,7 +78,7 @@ class Recommendation(OrderedEnum):
 
 
 class Proposal(NamedTuple):
-    candidates: Sequence[AlbumMatch | TrackMatch]
+    candidates: Sequence[Union[AlbumMatch, TrackMatch]]
```
Comment (snejus) [inline beets/autotag/match.py:81]: Forgot to mention - there's no need for this adjustment as it already utilizes the new pipe syntax.

Comment (amogus07) [reply beets/autotag/match.py:81]: it doesn‘t work below 3.10 though

Comment (snejus) [reply beets/autotag/match.py:81]: Try including `from __future__ import annotations` import at the top of the module

File: beets/ui/commands.py (hunk 4 above)
Comment (amogus07) [reply beets/ui/commands.py:667]: I think the person who wrote this would have to explain what it means…

Comment (amogus07) [reply beets/ui/commands.py:667]: @JOJ0 can you help out?

**Review by snejus — APPROVED** [review]
Thanks for making the requested adjustments, it all looks good now!
